// @dytesdk/ui-kit/dist/cjs/dyte-caption-toggle.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-1109742b.js');
const uiStore = require('./ui-store-c9fc6964.js');
const index = require('./index-42dce5ce.js');

const dyteCaptionToggleCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:block}";
const DyteCaptionToggleStyle0 = dyteCaptionToggleCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteCaptionToggle = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        this.stateUpdate = index$1.createEvent(this, "dyteStateUpdate", 7);
        /** Variant */
        this.variant = 'button';
        /** Config */
        this.config = uiStore.defaultConfig;
        /** Icon pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
        this.captionEnabled = false;
        this.permissionsUpdateListener = () => {
            var _a;
            this.captionEnabled =
                (_a = this.meeting.self.permissions.transcriptionEnabled) !== null && _a !== void 0 ? _a : false;
        };
    }
    connectedCallback() {
        this.meetingChanged(this.meeting);
    }
    meetingChanged(meeting) {
        if (meeting === null)
            return;
        this.permissionsUpdateListener();
        this.meeting.self.permissions.addListener('permissionsUpdate', this.permissionsUpdateListener);
    }
    disconnectedCallback() {
        var _a;
        (_a = this.meeting) === null || _a === void 0 ? void 0 : _a.self.permissions.removeListener('permissionsUpdate', this.permissionsUpdateListener);
    }
    toggleCaptions() {
        this.stateUpdate.emit({ activeCaptions: !this.states.activeCaptions, activeMoreMenu: false });
    }
    render() {
        if (!this.captionEnabled)
            return null;
        const captionsEnabled = this.states.activeCaptions;
        return (index$1.h(index$1.Host, { tabIndex: 0, role: "log", "aria-label": `Picture-in-Picture mode` }, index$1.h("dyte-controlbar-button", { part: "controlbar-button", size: this.size, iconPack: this.iconPack, onClick: () => this.toggleCaptions(), icon: captionsEnabled ? this.iconPack.captionsOff : this.iconPack.captionsOn, label: captionsEnabled ? this.t('transcript.off') : this.t('transcript.on'), variant: this.variant })));
    }
    static get watchers() { return {
        "meeting": ["meetingChanged"]
    }; }
};
__decorate([
    index.SyncWithStore()
], DyteCaptionToggle.prototype, "meeting", void 0);
__decorate([
    index.SyncWithStore()
], DyteCaptionToggle.prototype, "states", void 0);
__decorate([
    index.SyncWithStore()
], DyteCaptionToggle.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], DyteCaptionToggle.prototype, "size", void 0);
__decorate([
    index.SyncWithStore()
], DyteCaptionToggle.prototype, "t", void 0);
DyteCaptionToggle.style = DyteCaptionToggleStyle0;

exports.dyte_caption_toggle = DyteCaptionToggle;
