// @dytesdk/ui-kit/dist/cjs/dyte-waiting-screen.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-1109742b.js');
const uiStore = require('./ui-store-c9fc6964.js');
const index = require('./index-42dce5ce.js');

const dyteWaitingScreenCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{height:100%;width:100%;display:flex;flex-direction:column;align-items:center;justify-content:center}.centered{display:flex;flex-direction:column;align-items:center}dyte-logo{margin-bottom:var(--dyte-space-8, 32px);height:var(--dyte-space-12, 48px)}p{font-size:16px;border-radius:var(--dyte-border-radius-lg, 12px);--tw-bg-opacity:1;background-color:rgba(var(--dyte-colors-background-800, 30 30 30) / var(--tw-bg-opacity));padding-left:var(--dyte-space-8, 32px);padding-right:var(--dyte-space-8, 32px);padding-top:var(--dyte-space-4, 16px);padding-bottom:var(--dyte-space-4, 16px);color:rgb(var(--dyte-colors-text-1000, 255 255 255))}";
const DyteWaitingScreenStyle0 = dyteWaitingScreenCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteWaitingScreen = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        /** Config */
        this.config = uiStore.defaultConfig;
        /** Icon pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
    }
    render() {
        return (index$1.h(index$1.Host, { key: '7e6180f1f8736c28f6c8d0f759842074fc416728' }, index$1.h("slot", { key: '8ebb3469c20b95313774ff0b1c06f9d02d37362d' }, index$1.h("div", { key: 'e560017916361b842850493abfbf5967b4e61b4b', class: "centered", part: "content" }, index$1.h("dyte-logo", { key: '39005a67f41efa0d1a345c6208154877afd4d177', meeting: this.meeting, config: this.config, part: "logo", t: this.t }), index$1.h("p", { key: '144ae6b75e5f856759ca6aa35b30af1de75d398d' }, this.t('waitlist.body_text'))))));
    }
};
__decorate([
    index.SyncWithStore()
], DyteWaitingScreen.prototype, "meeting", void 0);
__decorate([
    index.SyncWithStore()
], DyteWaitingScreen.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], DyteWaitingScreen.prototype, "t", void 0);
DyteWaitingScreen.style = DyteWaitingScreenStyle0;

exports.dyte_waiting_screen = DyteWaitingScreen;
