// @dytesdk/ui-kit/dist/cjs/dyte-file-dropzone.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-1109742b.js');
const uiStore = require('./ui-store-c9fc6964.js');
const index = require('./index-42dce5ce.js');

const dyteFileDropzoneCss = "#dropzone{position:absolute;top:var(--dyte-space-0, 0px);right:var(--dyte-space-0, 0px);bottom:var(--dyte-space-0, 0px);left:var(--dyte-space-0, 0px);z-index:10;display:none;flex-direction:column;align-items:center;justify-content:center;--tw-bg-opacity:1;background-color:rgba(var(--dyte-colors-background-800, 30 30 30) / var(--tw-bg-opacity));color:rgb(var(--dyte-colors-text-700, 255 255 255 / 0.64))}#dropzone.active{display:flex;animation:0.2s slide-up ease-in}";
const DyteFileDropzoneStyle0 = dyteFileDropzoneCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteFileDropzone = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        this.onDropCallback = index$1.createEvent(this, "dropCallback", 7);
        /** Icon pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
        this.dropzoneActivated = false;
    }
    connectedCallback() {
        if (!this.hostEl)
            throw new Error('hostEl prop is required');
        this.hostEl.addEventListener('dragover', (e) => {
            e.preventDefault();
            this.dropzoneActivated = true;
        });
        this.hostEl.addEventListener('dragleave', () => {
            this.dropzoneActivated = false;
        });
        this.hostEl.addEventListener('drop', (e) => {
            e.preventDefault();
            this.dropzoneActivated = false;
            this.onDropCallback.emit(e);
        });
    }
    render() {
        return (index$1.h(index$1.Host, { key: '2c1ba91175328eb8d81c9de9b54a4ed5ff0d78e0' }, index$1.h("div", { key: '9a6d873ac4fbf439c0aade2313f13cad4c1168d2', id: "dropzone", class: { active: this.dropzoneActivated }, part: "dropzone" }, index$1.h("dyte-icon", { key: '7ea8fd7f990fe3a6353a631efa13895a46465692', icon: this.iconPack.attach }), index$1.h("p", { key: 'fa222c2ad9ec8874cb4bac0425ceb04cd396bb43' }, this.t('chat.send_attachment')))));
    }
};
__decorate([
    index.SyncWithStore()
], DyteFileDropzone.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], DyteFileDropzone.prototype, "t", void 0);
DyteFileDropzone.style = DyteFileDropzoneStyle0;

exports.dyte_file_dropzone = DyteFileDropzone;
