// @dytesdk/ui-kit/dist/cjs/dyte-header.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-1109742b.js');
const uiStore = require('./ui-store-c9fc6964.js');
const index$2 = require('./index-861582ac.js');
const index = require('./index-42dce5ce.js');

const dyteHeaderCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:flex;height:var(--dyte-space-12, 48px);align-items:center;--tw-bg-opacity:1;background-color:rgba(var(--dyte-colors-background-1000, 8 8 8) / var(--tw-bg-opacity));padding-left:var(--dyte-space-4, 16px);padding-right:var(--dyte-space-4, 16px)}@media only screen and (max-device-height: 480px) and (orientation: landscape){:host{display:none !important}}@media only screen and (max-height: 480px) and (orientation: landscape){:host{display:none !important}}";
const DyteHeaderStyle0 = dyteHeaderCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteHeader = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        /** Variant */
        this.variant = 'solid';
        /** Whether to render the default UI */
        this.disableRender = false;
        /** Config */
        this.config = uiStore.defaultConfig;
        /** Icon Pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
    }
    render() {
        const defaults = {
            meeting: this.meeting,
            config: this.config,
            states: this.states,
            t: this.t,
            iconPack: this.iconPack,
            size: this.size,
        };
        return (index$1.h(index$1.Host, { key: '8c6d62fc6a63020885d9e367b281d584e4f86580' }, !this.disableRender && index$1.h(index$2.Render, { key: '9e0ce3f2e2eb8bebe322591cedbb243b588636ce', element: "dyte-header", defaults: defaults, onlyChildren: true }), index$1.h("slot", { key: '7eef46d4d0c1bae48b3fe852cd8347aa33aa60dc' })));
    }
};
__decorate([
    index.SyncWithStore()
], DyteHeader.prototype, "meeting", void 0);
__decorate([
    index.SyncWithStore()
], DyteHeader.prototype, "states", void 0);
__decorate([
    index.SyncWithStore()
], DyteHeader.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], DyteHeader.prototype, "t", void 0);
__decorate([
    index.SyncWithStore()
], DyteHeader.prototype, "size", void 0);
DyteHeader.style = DyteHeaderStyle0;

exports.dyte_header = DyteHeader;
