// @dytesdk/ui-kit/dist/cjs/dyte-ui-provider.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index = require('./index-1109742b.js');
const uiStore = require('./ui-store-c9fc6964.js');
const config = require('./config-2a79f568.js');
const size = require('./size-2a406e36.js');
require('./keysIn-8bdbe83d.js');
require('./isObjectLike-73d13a39.js');

const dyteUiProviderCss = "dyte-ui-provider{display:block;width:100%;height:100%}";
const DyteUiProviderStyle0 = dyteUiProviderCss;

const LEAVE_ROOM_TIMER = 10000;
const DyteUiProvider = class {
    constructor(hostRef) {
        index.registerInstance(this, hostRef);
        this.statesUpdate = index.createEvent(this, "dyteStatesUpdate", 7);
        /** Config */
        this.config = uiStore.defaultConfig;
        /** Whether to show setup screen or not */
        this.showSetupScreen = false;
        /**
         * Do not render children until meeting is initialized
         * @default false
         */
        this.noRenderUntilMeeting = false;
        this.handleResize = () => {
            this.size = size.getSize(this.host.clientWidth);
        };
        this.loadTheme = () => {
            var _a;
            if (this.config === uiStore.defaultConfig) {
                const { config: config$1 } = config.generateConfig(this.meeting.self.config, this.meeting);
                this.config = config$1;
            }
            if ((_a = this.config) === null || _a === void 0 ? void 0 : _a.designTokens) {
                config.provideDyteDesignSystem(document.documentElement, this.config.designTokens);
            }
        };
        this.roomJoinedListener = () => {
            this.updateStates({ meeting: 'joined' });
        };
        this.waitlistedListener = () => {
            this.updateStates({ meeting: 'waiting' });
        };
        this.roomLeftListener = ({ state }) => {
            // Let socketConnectionUpdate listener handle this case.
            if (state === 'disconnected' || state === 'failed')
                return;
            this.updateStates({ meeting: 'ended', roomLeftState: state });
        };
        this.mediaPermissionUpdateListener = ({ kind, message }) => {
            if (['audio', 'video'].includes(kind)) {
                if ((message === 'DENIED' || message === 'SYSTEM_DENIED') &&
                    uiStore.uiState.states.activeDebugger !== true) {
                    const permissionModalSettings = {
                        enabled: true,
                        kind,
                    };
                    this.updateStates({ activePermissionsMessage: permissionModalSettings });
                }
            }
        };
        this.socketConnectionUpdateListener = ({ state }) => {
            if (state === 'failed') {
                setTimeout(() => {
                    this.meeting.leave('disconnected');
                }, LEAVE_ROOM_TIMER);
            }
        };
        this.handleChangingMeeting = (destinationMeetingId) => {
            this.updateStates({
                activeBreakoutRoomsManager: Object.assign(Object.assign({}, uiStore.uiState.states.activeBreakoutRoomsManager), { destinationMeetingId }),
            });
        };
    }
    listenState(e) {
        this.updateStates(e.detail);
    }
    updateStates(states) {
        const newStates = Object.assign({}, uiStore.uiState.states);
        uiStore.uiState.states = config.deepMerge(newStates, states);
        this.statesUpdate.emit(uiStore.uiState.states);
    }
    connectedCallback() {
        if (typeof window !== 'undefined') {
            this.authErrorListener = (ev) => {
                if (ev.detail.message.includes('401')) {
                    this.updateStates({ meeting: 'ended', roomLeftState: 'unauthorized' });
                }
            };
            window.addEventListener('dyteError', this.authErrorListener);
        }
        this.onMeetingChange(this.meeting);
        this.onIconPackChange(this.iconPack);
        this.onTChange(this.t);
        this.onConfigChange(this.config);
        this.onSizeChange(this.size);
        this.resizeObserver = new ResizeObserver(() => this.handleResize());
        this.resizeObserver.observe(this.host);
    }
    disconnectedCallback() {
        this.resizeObserver.disconnect();
        window.removeEventListener('dyteError', this.authErrorListener);
        if (!this.meeting)
            return;
        this.meeting.self.removeListener('roomLeft', this.roomLeftListener);
        this.meeting.self.removeListener('roomJoined', this.roomJoinedListener);
        this.meeting.self.removeListener('waitlisted', this.waitlistedListener);
        this.meeting.self.removeListener('mediaPermissionUpdate', this.mediaPermissionUpdateListener);
        this.meeting.meta.removeListener('socketConnectionUpdate', this.socketConnectionUpdateListener);
    }
    onMeetingChange(meeting) {
        uiStore.uiStore.state.meeting = meeting;
        if (!meeting)
            return;
        this.updateStates({ viewType: meeting.meta.viewType });
        this.loadTheme();
        meeting.self.addListener('roomJoined', this.roomJoinedListener);
        meeting.self.addListener('waitlisted', this.waitlistedListener);
        meeting.self.addListener('roomLeft', this.roomLeftListener);
        meeting.self.addListener('mediaPermissionUpdate', this.mediaPermissionUpdateListener);
        meeting.meta.addListener('socketConnectionUpdate', this.socketConnectionUpdateListener);
        if (meeting.connectedMeetings.supportsConnectedMeetings) {
            meeting.connectedMeetings.once('changingMeeting', this.handleChangingMeeting);
        }
        if (meeting.self.roomJoined) {
            this.updateStates({ meeting: 'joined' });
        }
        else {
            if (this.showSetupScreen) {
                this.updateStates({ meeting: 'setup' });
            }
            else {
                meeting.joinRoom();
            }
        }
        window.removeEventListener('dyteError', this.authErrorListener);
    }
    onIconPackChange(newIconPack) {
        uiStore.uiStore.state.iconPack = newIconPack;
    }
    onTChange(newT) {
        uiStore.uiStore.state.t = newT;
    }
    onConfigChange(config) {
        uiStore.uiStore.state.config = config;
    }
    onSizeChange(newSize) {
        uiStore.uiStore.state.size = newSize;
    }
    render() {
        return index.h(index.Host, { key: 'aa4172c4b0395a5262697346a2731c1829edb369' }, this.noRenderUntilMeeting && !this.meeting ? null : index.h("slot", null));
    }
    get host() { return index.getElement(this); }
    static get watchers() { return {
        "meeting": ["onMeetingChange"],
        "iconPack": ["onIconPackChange"],
        "t": ["onTChange"],
        "config": ["onConfigChange"],
        "size": ["onSizeChange"]
    }; }
};
DyteUiProvider.style = DyteUiProviderStyle0;

exports.dyte_ui_provider = DyteUiProvider;
