// @dytesdk/ui-kit/dist/cjs/dyte-controlbar.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-1109742b.js');
const uiStore = require('./ui-store-c9fc6964.js');
const index$2 = require('./index-861582ac.js');
const index = require('./index-42dce5ce.js');

const dyteControlbarCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{box-sizing:border-box;display:flex;align-items:center;gap:var(--dyte-space-0\\.5, 2px);--tw-bg-opacity:1;background-color:rgba(var(--dyte-colors-background-1000, 8 8 8) / var(--tw-bg-opacity));padding-left:var(--dyte-space-4, 16px);padding-right:var(--dyte-space-4, 16px);position:relative;z-index:10}@media only screen and (max-device-height: 480px) and (orientation: landscape){:host{padding-top:var(--dyte-space-0, 0px) !important}}";
const DyteControlbarStyle0 = dyteControlbarCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteControlbar = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        /** Variant */
        this.variant = 'solid';
        /** Whether to render the default UI */
        this.disableRender = false;
        /** Config */
        this.config = uiStore.defaultConfig;
        /** Icon Pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
    }
    render() {
        const defaults = {
            meeting: this.meeting,
            config: this.config,
            states: this.states,
            t: this.t,
            iconPack: this.iconPack,
            size: this.size,
        };
        return (index$1.h(index$1.Host, { key: 'ef0ba36f97e1366fbeab0d7e5d0eeab4119e898c' }, !this.disableRender && (index$1.h(index$2.Render, { key: '2f8951fb89ebb0d81bc53eb3eeebd8c3de074ed9', element: "dyte-controlbar", defaults: defaults, onlyChildren: true })), index$1.h("slot", { key: 'd9e094a525ccffdd485eb6de23920c05c36c6ed1' })));
    }
};
__decorate([
    index.SyncWithStore()
], DyteControlbar.prototype, "meeting", void 0);
__decorate([
    index.SyncWithStore()
], DyteControlbar.prototype, "states", void 0);
__decorate([
    index.SyncWithStore()
], DyteControlbar.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], DyteControlbar.prototype, "t", void 0);
__decorate([
    index.SyncWithStore()
], DyteControlbar.prototype, "size", void 0);
DyteControlbar.style = DyteControlbarStyle0;

exports.dyte_controlbar = DyteControlbar;
