// @dytesdk/ui-kit/dist/cjs/dyte-idle-screen.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const index$1 = require('./index-1109742b.js');
const uiStore = require('./ui-store-c9fc6964.js');
const index = require('./index-42dce5ce.js');

const dyteIdleScreenCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{height:100%;width:100%;display:flex;flex-direction:column;align-items:center;justify-content:center}.ctr{display:flex;flex-direction:column;align-items:center;gap:var(--dyte-space-8, 32px)}dyte-logo.loaded{height:var(--dyte-space-12, 48px)}dyte-spinner{height:var(--dyte-space-12, 48px);width:var(--dyte-space-12, 48px);--tw-text-opacity:1;color:rgba(var(--dyte-colors-brand-500, 33 96 253) / var(--tw-text-opacity))}";
const DyteIdleScreenStyle0 = dyteIdleScreenCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteIdleScreen = class {
    constructor(hostRef) {
        index$1.registerInstance(this, hostRef);
        /** Config object */
        this.config = uiStore.defaultConfig;
        /** Icon pack */
        this.iconPack = uiStore.defaultIconPack;
        /** Language */
        this.t = uiStore.useLanguage();
    }
    render() {
        return (index$1.h(index$1.Host, { key: '12d044e448e70b95bbaadb456a59ae322cbec4cb' }, index$1.h("slot", { key: 'befccedc79773180014dbd0fc9b944055f888e3d' }, index$1.h("div", { key: '98c261026b266fbb8b8bdbc143aefc390f3adeb0', class: "ctr", part: "container" }, index$1.h("dyte-logo", { key: '900a6cc8979009204f29b31908591319e00952ad', meeting: this.meeting, config: this.config, t: this.t, part: "logo" }), index$1.h("dyte-spinner", { key: 'a9575923df832e8af9b130eee2dbd0e8fc42e181', "aria-label": "Idle, waiting for meeting data", part: "spinner", iconPack: this.iconPack })))));
    }
};
__decorate([
    index.SyncWithStore()
], DyteIdleScreen.prototype, "meeting", void 0);
__decorate([
    index.SyncWithStore()
], DyteIdleScreen.prototype, "iconPack", void 0);
__decorate([
    index.SyncWithStore()
], DyteIdleScreen.prototype, "t", void 0);
DyteIdleScreen.style = DyteIdleScreenStyle0;

exports.dyte_idle_screen = DyteIdleScreen;
