// @dytesdk/ui-kit/dist/components/dyte-waiting-screen.js
import { p as proxyCustomElement, H, h, e as Host } from './p-1d5392d7.js';
import { d as defaultConfig, a as defaultIconPack, c as useLanguage } from './p-dce5d30e.js';
import './p-e497ad6d.js';
import { S as SyncWithStore } from './p-9ae68517.js';
import { d as defineCustomElement$2 } from './p-9740eba3.js';

const dyteWaitingScreenCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{height:100%;width:100%;display:flex;flex-direction:column;align-items:center;justify-content:center}.centered{display:flex;flex-direction:column;align-items:center}dyte-logo{margin-bottom:var(--dyte-space-8, 32px);height:var(--dyte-space-12, 48px)}p{font-size:16px;border-radius:var(--dyte-border-radius-lg, 12px);--tw-bg-opacity:1;background-color:rgba(var(--dyte-colors-background-800, 30 30 30) / var(--tw-bg-opacity));padding-left:var(--dyte-space-8, 32px);padding-right:var(--dyte-space-8, 32px);padding-top:var(--dyte-space-4, 16px);padding-bottom:var(--dyte-space-4, 16px);color:rgb(var(--dyte-colors-text-1000, 255 255 255))}";
const DyteWaitingScreenStyle0 = dyteWaitingScreenCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteWaitingScreen$1 = /*@__PURE__*/ proxyCustomElement(class DyteWaitingScreen extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Config */
        this.config = defaultConfig;
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
    }
    render() {
        return (h(Host, { key: '7e6180f1f8736c28f6c8d0f759842074fc416728' }, h("slot", { key: '8ebb3469c20b95313774ff0b1c06f9d02d37362d' }, h("div", { key: 'e560017916361b842850493abfbf5967b4e61b4b', class: "centered", part: "content" }, h("dyte-logo", { key: '39005a67f41efa0d1a345c6208154877afd4d177', meeting: this.meeting, config: this.config, part: "logo", t: this.t }), h("p", { key: '144ae6b75e5f856759ca6aa35b30af1de75d398d' }, this.t('waitlist.body_text'))))));
    }
    static get style() { return DyteWaitingScreenStyle0; }
}, [1, "dyte-waiting-screen", {
        "meeting": [16],
        "config": [16],
        "iconPack": [16],
        "t": [16]
    }]);
__decorate([
    SyncWithStore()
], DyteWaitingScreen$1.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteWaitingScreen$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteWaitingScreen$1.prototype, "t", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-waiting-screen", "dyte-logo"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-waiting-screen":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteWaitingScreen$1);
            }
            break;
        case "dyte-logo":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}
defineCustomElement$1();

const DyteWaitingScreen = DyteWaitingScreen$1;
const defineCustomElement = defineCustomElement$1;

export { DyteWaitingScreen, defineCustomElement };
