// @dytesdk/ui-kit/dist/components/dyte-file-dropzone.js
import { p as proxyCustomElement, H, d as createEvent, h, e as Host } from './p-1d5392d7.js';
import { a as defaultIconPack, c as useLanguage } from './p-dce5d30e.js';
import './p-e497ad6d.js';
import { S as SyncWithStore } from './p-9ae68517.js';
import { d as defineCustomElement$2 } from './p-a443344b.js';

const dyteFileDropzoneCss = "#dropzone{position:absolute;top:var(--dyte-space-0, 0px);right:var(--dyte-space-0, 0px);bottom:var(--dyte-space-0, 0px);left:var(--dyte-space-0, 0px);z-index:10;display:none;flex-direction:column;align-items:center;justify-content:center;--tw-bg-opacity:1;background-color:rgba(var(--dyte-colors-background-800, 30 30 30) / var(--tw-bg-opacity));color:rgb(var(--dyte-colors-text-700, 255 255 255 / 0.64))}#dropzone.active{display:flex;animation:0.2s slide-up ease-in}";
const DyteFileDropzoneStyle0 = dyteFileDropzoneCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteFileDropzone$1 = /*@__PURE__*/ proxyCustomElement(class DyteFileDropzone extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.onDropCallback = createEvent(this, "dropCallback", 7);
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        this.dropzoneActivated = false;
    }
    connectedCallback() {
        if (!this.hostEl)
            throw new Error('hostEl prop is required');
        this.hostEl.addEventListener('dragover', (e) => {
            e.preventDefault();
            this.dropzoneActivated = true;
        });
        this.hostEl.addEventListener('dragleave', () => {
            this.dropzoneActivated = false;
        });
        this.hostEl.addEventListener('drop', (e) => {
            e.preventDefault();
            this.dropzoneActivated = false;
            this.onDropCallback.emit(e);
        });
    }
    render() {
        return (h(Host, { key: '2c1ba91175328eb8d81c9de9b54a4ed5ff0d78e0' }, h("div", { key: '9a6d873ac4fbf439c0aade2313f13cad4c1168d2', id: "dropzone", class: { active: this.dropzoneActivated }, part: "dropzone" }, h("dyte-icon", { key: '7ea8fd7f990fe3a6353a631efa13895a46465692', icon: this.iconPack.attach }), h("p", { key: 'fa222c2ad9ec8874cb4bac0425ceb04cd396bb43' }, this.t('chat.send_attachment')))));
    }
    static get style() { return DyteFileDropzoneStyle0; }
}, [1, "dyte-file-dropzone", {
        "hostEl": [16],
        "iconPack": [16],
        "t": [16],
        "dropzoneActivated": [32]
    }]);
__decorate([
    SyncWithStore()
], DyteFileDropzone$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteFileDropzone$1.prototype, "t", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-file-dropzone", "dyte-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-file-dropzone":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteFileDropzone$1);
            }
            break;
        case "dyte-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}
defineCustomElement$1();

const DyteFileDropzone = DyteFileDropzone$1;
const defineCustomElement = defineCustomElement$1;

export { DyteFileDropzone, defineCustomElement };
