// @dytesdk/ui-kit/dist/components/dyte-simple-grid.js
import { p as proxyCustomElement, H, h, e as Host } from './p-1d5392d7.js';
import { d as defaultConfig, a as defaultIconPack, c as useLanguage } from './p-dce5d30e.js';
import { R as Render } from './p-52218cb0.js';
import { u as useGrid } from './p-b13ddb7d.js';
import { i as index } from './p-f47d4fe8.js';
import { S as SyncWithStore } from './p-9ae68517.js';

const dyteSimpleGridCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{position:relative;display:block;height:100%;width:100%}";
const DyteSimpleGridStyle0 = dyteSimpleGridCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteSimpleGrid$1 = /*@__PURE__*/ proxyCustomElement(class DyteSimpleGrid extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Participants */
        this.participants = [];
        /**
         * Aspect Ratio of participant tile
         *
         * Format: `width:height`
         */
        this.aspectRatio = '16:9';
        /** Gap between participant tiles */
        this.gap = 8;
        /** UI Config */
        this.config = defaultConfig;
        /** Icon Pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        this.dimensions = { width: 0, height: 0 };
        this.setHostDimensions = () => {
            const { clientWidth: width, clientHeight: height } = this.host;
            this.dimensions = { width, height };
        };
    }
    connectedCallback() {
        this.resizeObserver = new index(this.setHostDimensions);
        this.resizeObserver.observe(this.host);
        this.meetingChanged(this.meeting);
    }
    meetingChanged(meeting) {
        const meta = meeting === null || meeting === void 0 ? void 0 : meeting.meta;
        if (meta)
            this.mediaConnection = Object.assign({}, meta.mediaState);
    }
    disconnectedCallback() {
        var _a;
        (_a = this.resizeObserver) === null || _a === void 0 ? void 0 : _a.disconnect();
    }
    render() {
        const defaults = {
            meeting: this.meeting,
            config: this.config,
            states: this.states,
            size: this.size,
            iconPack: this.iconPack,
            t: this.t,
        };
        const { width, height, getPosition } = useGrid({
            dimensions: this.dimensions,
            count: this.participants.length,
            aspectRatio: this.aspectRatio,
            gap: this.gap,
        });
        return (h(Host, { key: '966acd8e36eb46608d9dc7aa026497b86b062be5' }, h("slot", { key: 'b67280589e7b5ecd630da8584ea34755400c9a33' }, this.participants.map((participant, index) => {
            const { top, left } = getPosition(index);
            return (h(Render, { element: "dyte-participant-tile", defaults: defaults, props: {
                    participant,
                    style: {
                        position: 'absolute',
                        top: `${top}px`,
                        left: `${left}px`,
                        width: `${width}px`,
                        height: `${height}px`,
                    },
                    key: participant.id,
                    'data-participant': participant.id,
                    mediaConnection: this.mediaConnection,
                }, childProps: { participant }, deepProps: true }));
        }))));
    }
    get host() { return this; }
    static get watchers() { return {
        "meeting": ["meetingChanged"]
    }; }
    static get style() { return DyteSimpleGridStyle0; }
}, [1, "dyte-simple-grid", {
        "participants": [16],
        "aspectRatio": [1, "aspect-ratio"],
        "gap": [2],
        "size": [513],
        "meeting": [16],
        "states": [16],
        "config": [16],
        "iconPack": [16],
        "t": [16],
        "dimensions": [32],
        "mediaConnection": [32]
    }, undefined, {
        "meeting": ["meetingChanged"]
    }]);
__decorate([
    SyncWithStore()
], DyteSimpleGrid$1.prototype, "size", void 0);
__decorate([
    SyncWithStore()
], DyteSimpleGrid$1.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteSimpleGrid$1.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], DyteSimpleGrid$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteSimpleGrid$1.prototype, "t", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-simple-grid"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-simple-grid":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteSimpleGrid$1);
            }
            break;
    } });
}
defineCustomElement$1();

const DyteSimpleGrid = DyteSimpleGrid$1;
const defineCustomElement = defineCustomElement$1;

export { DyteSimpleGrid, defineCustomElement };
