// @dytesdk/ui-kit/dist/components/dyte-stage.js
import { p as proxyCustomElement, H, d as createEvent, h, e as Host } from './p-1d5392d7.js';
import { a as defaultIconPack, c as useLanguage } from './p-dce5d30e.js';
import { S as SyncWithStore } from './p-9ae68517.js';

const dyteStageCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:block;position:relative;overflow:hidden}";
const DyteStageStyle0 = dyteStageCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteStage$1 = /*@__PURE__*/ proxyCustomElement(class DyteStage extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.stateUpdate = createEvent(this, "dyteStateUpdate", 7);
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
    }
    render() {
        return (h(Host, { key: '4d2ff05bef43b31e74d221c715bf6e53e32b641e' }, h("slot", { key: '55a21b831c9571a773707488398e113d135b6b1d' })));
    }
    static get style() { return DyteStageStyle0; }
}, [1, "dyte-stage", {
        "iconPack": [16],
        "t": [16]
    }]);
__decorate([
    SyncWithStore()
], DyteStage$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteStage$1.prototype, "t", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-stage"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-stage":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteStage$1);
            }
            break;
    } });
}
defineCustomElement$1();

const DyteStage = DyteStage$1;
const defineCustomElement = defineCustomElement$1;

export { DyteStage, defineCustomElement };
