// @dytesdk/ui-kit/dist/components/dyte-meeting-title.js
import { p as proxyCustomElement, H, h, e as Host } from './p-1d5392d7.js';
import { a as defaultIconPack, c as useLanguage } from './p-dce5d30e.js';
import { S as SyncWithStore } from './p-9ae68517.js';
import { d as defineCustomElement$2 } from './p-b4f8107f.js';

const dyteMeetingTitleCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{margin-left:var(--dyte-space-3, 12px);margin-right:var(--dyte-space-3, 12px);-webkit-user-select:none;-moz-user-select:none;user-select:none;font-size:16px;color:rgb(var(--dyte-colors-text-1000, 255 255 255))}.title{text-align:center;overflow:hidden;display:-webkit-box;-webkit-box-orient:vertical;-webkit-line-clamp:1}:host([size='sm']){font-size:12px}";
const DyteMeetingTitleStyle0 = dyteMeetingTitleCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteMeetingTitle$1 = /*@__PURE__*/ proxyCustomElement(class DyteMeetingTitle extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
    }
    render() {
        var _a;
        const title = (_a = this.meeting) === null || _a === void 0 ? void 0 : _a.meta.meetingTitle;
        if (title == null)
            return null;
        return (h(Host, { tabIndex: 0, role: "banner", "aria-label": title }, h("dyte-tooltip", { label: title, part: "tooltip" }, h("div", { class: "title", part: "title" }, title))));
    }
    static get style() { return DyteMeetingTitleStyle0; }
}, [1, "dyte-meeting-title", {
        "meeting": [16],
        "iconPack": [16],
        "t": [16]
    }]);
__decorate([
    SyncWithStore()
], DyteMeetingTitle$1.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteMeetingTitle$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteMeetingTitle$1.prototype, "t", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-meeting-title", "dyte-tooltip"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-meeting-title":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteMeetingTitle$1);
            }
            break;
        case "dyte-tooltip":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}
defineCustomElement$1();

const DyteMeetingTitle = DyteMeetingTitle$1;
const defineCustomElement = defineCustomElement$1;

export { DyteMeetingTitle, defineCustomElement };
