// @dytesdk/ui-kit/dist/components/dyte-spotlight-grid.js
import { p as proxyCustomElement, H, h, e as Host } from './p-1d5392d7.js';
import { d as defaultConfig, a as defaultIconPack, c as useLanguage } from './p-dce5d30e.js';
import { d as defaultGridSize } from './p-b13ddb7d.js';
import { R as Render } from './p-52218cb0.js';
import { S as SyncWithStore } from './p-9ae68517.js';

const dyteSpotlightGridCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:flex;height:100%;width:100%}main{flex:1 1 0%}.grid-width-sm{width:25%}.grid-width-md{width:50%}.grid-width-lg{width:66.666667%}:host([size='sm']),:host([size='md']){flex-direction:column}:host([size='md']) aside{max-height:var(--dyte-space-96, 384px);width:100%;max-width:100%}:host([size='md']) .grid-width-sm{height:25%}:host([size='md']) .grid-width-md{height:50%}:host([size='md']) .grid-width-lg{height:66.666667%}:host([size='sm']) aside{max-height:var(--dyte-space-96, 384px);width:100%;max-width:100%}:host([size='sm']) .grid-width-sm,:host([size='sm']) .grid-width-md,:host([size='sm']) .grid-width-lg{height:50%}:host([size='xl']) .grid-width-sm,:host([size='xl']) .grid-width-md,:host([size='xl']) .grid-width-lg{width:400px}:host([layout='column']){flex-direction:column}:host([layout='column']) main{flex:4}:host([layout='column']) aside{flex:2;max-width:100%;width:100%}";
const DyteSpotlightGridStyle0 = dyteSpotlightGridCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteSpotlightGrid$1 = /*@__PURE__*/ proxyCustomElement(class DyteSpotlightGrid extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Grid Layout */
        this.layout = 'row';
        /** Participants */
        this.participants = [];
        /** Pinned Participants */
        this.pinnedParticipants = [];
        /**
         * Aspect Ratio of participant tile
         *
         * Format: `width:height`
         */
        this.aspectRatio = '16:9';
        /** Gap between participant tiles */
        this.gap = 8;
        /** UI Config */
        this.config = defaultConfig;
        /** Icon Pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        /** Grid size */
        this.gridSize = defaultGridSize;
        this.getAdaptiveSize = (length) => {
            if (this.size === 'sm') {
                return 'sm';
            }
            if (length > 3) {
                return 'sm';
            }
            else {
                if (this.size === 'md') {
                    return 'sm';
                }
                return 'md';
            }
        };
    }
    render() {
        const defaults = {
            meeting: this.meeting,
            config: this.config,
            states: this.states,
            size: this.size,
            iconPack: this.iconPack,
            t: this.t,
        };
        const nonPinnedParticipants = this.participants.filter((p) => this.pinnedParticipants.some((pt) => pt.id != p.id));
        return (h(Host, { key: 'dbae02de6986b50f91073c0d3e397b0bba1b8dae' }, h("main", { key: '3aab79ea8fb10a418c7e128b7825baf6ef9b9ccf', part: "main" }, h(Render, { key: 'a416ee3b0eef6fc99cf04bce32d88a26ee3b50c9', element: "dyte-simple-grid", defaults: defaults, props: {
                part: 'main-grid',
                participants: this.pinnedParticipants,
                aspectRatio: this.aspectRatio,
                gap: this.gap,
                size: this.getAdaptiveSize(this.pinnedParticipants.length),
            } })), nonPinnedParticipants.length > 0 && (h("aside", { key: '266b02b5d0737b31c573028fef1d11b820ac7e39', part: "aside", class: this.gridSize.spotlight ? `grid-width-${this.gridSize.spotlight}` : 'grid-width-md' }, h(Render, { key: '283f6a3f085291b1b450f21f39459953a3423915', element: "dyte-simple-grid", defaults: defaults, props: {
                part: 'aside-grid',
                participants: nonPinnedParticipants,
                aspectRatio: this.aspectRatio,
                gap: this.gap,
                size: this.getAdaptiveSize(nonPinnedParticipants.length),
            } })))));
    }
    get host() { return this; }
    static get style() { return DyteSpotlightGridStyle0; }
}, [1, "dyte-spotlight-grid", {
        "layout": [513],
        "participants": [16],
        "pinnedParticipants": [16],
        "aspectRatio": [1, "aspect-ratio"],
        "gap": [2],
        "size": [513],
        "meeting": [16],
        "states": [16],
        "config": [16],
        "iconPack": [16],
        "t": [16],
        "gridSize": [16]
    }]);
__decorate([
    SyncWithStore()
], DyteSpotlightGrid$1.prototype, "size", void 0);
__decorate([
    SyncWithStore()
], DyteSpotlightGrid$1.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteSpotlightGrid$1.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], DyteSpotlightGrid$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteSpotlightGrid$1.prototype, "t", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-spotlight-grid"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-spotlight-grid":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteSpotlightGrid$1);
            }
            break;
    } });
}
defineCustomElement$1();

const DyteSpotlightGrid = DyteSpotlightGrid$1;
const defineCustomElement = defineCustomElement$1;

export { DyteSpotlightGrid, defineCustomElement };
