// @dytesdk/ui-kit/dist/components/dyte-controlbar.js
import { p as proxyCustomElement, H, h, e as Host } from './p-1d5392d7.js';
import { d as defaultConfig, a as defaultIconPack, c as useLanguage } from './p-dce5d30e.js';
import './p-e497ad6d.js';
import { R as Render } from './p-52218cb0.js';
import { S as SyncWithStore } from './p-9ae68517.js';

const dyteControlbarCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{box-sizing:border-box;display:flex;align-items:center;gap:var(--dyte-space-0\\.5, 2px);--tw-bg-opacity:1;background-color:rgba(var(--dyte-colors-background-1000, 8 8 8) / var(--tw-bg-opacity));padding-left:var(--dyte-space-4, 16px);padding-right:var(--dyte-space-4, 16px);position:relative;z-index:10}@media only screen and (max-device-height: 480px) and (orientation: landscape){:host{padding-top:var(--dyte-space-0, 0px) !important}}";
const DyteControlbarStyle0 = dyteControlbarCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteControlbar$1 = /*@__PURE__*/ proxyCustomElement(class DyteControlbar extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Variant */
        this.variant = 'solid';
        /** Whether to render the default UI */
        this.disableRender = false;
        /** Config */
        this.config = defaultConfig;
        /** Icon Pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
    }
    render() {
        const defaults = {
            meeting: this.meeting,
            config: this.config,
            states: this.states,
            t: this.t,
            iconPack: this.iconPack,
            size: this.size,
        };
        return (h(Host, { key: 'ef0ba36f97e1366fbeab0d7e5d0eeab4119e898c' }, !this.disableRender && (h(Render, { key: '2f8951fb89ebb0d81bc53eb3eeebd8c3de074ed9', element: "dyte-controlbar", defaults: defaults, onlyChildren: true })), h("slot", { key: 'd9e094a525ccffdd485eb6de23920c05c36c6ed1' })));
    }
    static get style() { return DyteControlbarStyle0; }
}, [1, "dyte-controlbar", {
        "variant": [513],
        "disableRender": [4, "disable-render"],
        "meeting": [16],
        "config": [16],
        "states": [16],
        "iconPack": [16],
        "t": [16],
        "size": [513]
    }]);
__decorate([
    SyncWithStore()
], DyteControlbar$1.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteControlbar$1.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], DyteControlbar$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteControlbar$1.prototype, "t", void 0);
__decorate([
    SyncWithStore()
], DyteControlbar$1.prototype, "size", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-controlbar"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-controlbar":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteControlbar$1);
            }
            break;
    } });
}
defineCustomElement$1();

const DyteControlbar = DyteControlbar$1;
const defineCustomElement = defineCustomElement$1;

export { DyteControlbar, defineCustomElement };
