// @dytesdk/ui-kit/dist/components/dyte-livestream-toggle.js
import { p as proxyCustomElement, H, d as createEvent, h, e as Host } from './p-1d5392d7.js';
import { a as defaultIconPack, c as useLanguage, i as isLiveStreamHost } from './p-dce5d30e.js';
import { S as SyncWithStore } from './p-9ae68517.js';
import { d as defineCustomElement$4 } from './p-9e83e8fe.js';
import { d as defineCustomElement$3 } from './p-a443344b.js';
import { d as defineCustomElement$2 } from './p-cba97631.js';

const dyteLivestreamToggleCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:block}:host([data-hidden]){display:none}";
const DyteLivestreamToggleStyle0 = dyteLivestreamToggleCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteLivestreamToggle$1 = /*@__PURE__*/ proxyCustomElement(class DyteLivestreamToggle extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.stateUpdate = createEvent(this, "dyteStateUpdate", 7);
        this.dyteAPIError = createEvent(this, "dyteAPIError", 7);
        /** Variant */
        this.variant = 'button';
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        /** Is Livestream active */
        this.livestreamState = 'IDLE';
        this.livestreamStateListener = (state) => {
            this.livestreamState = state;
            if (state === 'LIVESTREAMING' || state === 'IDLE') {
                this.stateUpdate.emit({ activeMoreMenu: false });
            }
        };
        this.isLoading = () => {
            return (this.meeting == null ||
                this.livestreamState === 'STARTING' ||
                this.livestreamState === 'STOPPING');
        };
    }
    connectedCallback() {
        this.meetingChanged(this.meeting);
    }
    disconnectedCallback() {
        this.clearListeners();
    }
    meetingChanged(meeting) {
        var _a, _b;
        if (meeting == null)
            return;
        this.livestreamState = (_a = this.meeting.livestream) === null || _a === void 0 ? void 0 : _a.state;
        (_b = this.meeting.livestream) === null || _b === void 0 ? void 0 : _b.on('livestreamUpdate', this.livestreamStateListener);
    }
    async toggleLivestream() {
        var _a, _b;
        if (this.livestreamState === 'LIVESTREAMING') {
            try {
                await ((_a = this.meeting.livestream) === null || _a === void 0 ? void 0 : _a.stop());
            }
            catch (_c) {
                this.dyteAPIError.emit({
                    trace: this.t('livestream.stop'),
                    message: this.t('livestream.error.stop'),
                });
            }
        }
        else {
            try {
                await ((_b = this.meeting.livestream) === null || _b === void 0 ? void 0 : _b.start());
            }
            catch (_d) {
                this.dyteAPIError.emit({
                    trace: this.t('livestream.start'),
                    message: this.t('livestream.error.start'),
                });
            }
        }
    }
    clearListeners() {
        var _a;
        (_a = this.meeting.livestream) === null || _a === void 0 ? void 0 : _a.removeListener('livestreamUpdate', this.livestreamStateListener);
    }
    getLivestreamLabel() {
        switch (this.livestreamState) {
            case 'IDLE':
                return this.t('livestream.go_live');
            case 'LIVESTREAMING':
                return this.t('livestream.end_live');
            case 'WAITING_ON_MANUAL_INGESTION':
                return this.t('livestream.waiting_on_manual_ingestion');
            case 'STARTING':
                return this.t('livestream.starting');
            case 'STOPPING':
                return this.t('livestream.stopping');
            default:
                return this.t('livestream.error');
        }
    }
    getLivestreamIcon() {
        switch (this.livestreamState) {
            case 'IDLE':
                return this.iconPack.start_livestream;
            case 'LIVESTREAMING':
                return this.iconPack.stop_livestream;
            case 'WAITING_ON_MANUAL_INGESTION':
                return this.iconPack.start_livestream;
            case 'STARTING':
            case 'STOPPING':
            default:
                return this.iconPack.stop_livestream;
        }
    }
    render() {
        if (!isLiveStreamHost(this.meeting))
            return h(Host, { "data-hidden": true });
        return (h(Host, null, h("dyte-controlbar-button", { part: "controlbar-button", size: this.size, iconPack: this.iconPack, isLoading: this.isLoading(), class: { 'active-livestream': this.livestreamState === 'LIVESTREAMING' }, onClick: () => this.toggleLivestream(), icon: this.getLivestreamIcon(), disabled: this.isLoading(), label: this.t(this.getLivestreamLabel()), variant: this.variant })));
    }
    static get watchers() { return {
        "meeting": ["meetingChanged"]
    }; }
    static get style() { return DyteLivestreamToggleStyle0; }
}, [1, "dyte-livestream-toggle", {
        "variant": [513],
        "meeting": [16],
        "size": [513],
        "iconPack": [16],
        "t": [16],
        "livestreamState": [32]
    }, undefined, {
        "meeting": ["meetingChanged"]
    }]);
__decorate([
    SyncWithStore()
], DyteLivestreamToggle$1.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteLivestreamToggle$1.prototype, "size", void 0);
__decorate([
    SyncWithStore()
], DyteLivestreamToggle$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteLivestreamToggle$1.prototype, "t", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-livestream-toggle", "dyte-controlbar-button", "dyte-icon", "dyte-spinner"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-livestream-toggle":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteLivestreamToggle$1);
            }
            break;
        case "dyte-controlbar-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dyte-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dyte-spinner":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}
defineCustomElement$1();

const DyteLivestreamToggle = DyteLivestreamToggle$1;
const defineCustomElement = defineCustomElement$1;

export { DyteLivestreamToggle, defineCustomElement };
