// @dytesdk/ui-kit/dist/components/dyte-debugger-toggle.js
import { p as proxyCustomElement, H, d as createEvent, h, e as Host } from './p-1d5392d7.js';
import { c as useLanguage, a as defaultIconPack } from './p-dce5d30e.js';
import { S as SyncWithStore } from './p-9ae68517.js';
import { d as defineCustomElement$4 } from './p-9e83e8fe.js';
import { d as defineCustomElement$3 } from './p-a443344b.js';
import { d as defineCustomElement$2 } from './p-cba97631.js';

const dyteDebuggerToggleCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:block}";
const DyteDebuggerToggleStyle0 = dyteDebuggerToggleCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteDebuggerToggle$1 = /*@__PURE__*/ proxyCustomElement(class DyteDebuggerToggle extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.stateUpdate = createEvent(this, "dyteStateUpdate", 7);
        /** Variant */
        this.variant = 'button';
        /** Language */
        this.t = useLanguage();
        /** Icon pack */
        this.iconPack = defaultIconPack;
    }
    toggleDebugger() {
        var _a;
        this.stateUpdate.emit({
            activeDebugger: !((_a = this.states) === null || _a === void 0 ? void 0 : _a.activeDebugger),
            activeMoreMenu: false,
        });
    }
    render() {
        return (h(Host, { key: 'e4f3441ebff39ea51b49aa9a015b6875b433b7dd', title: this.t('Troubleshooting') }, h("dyte-controlbar-button", { key: '7441cd4adf6a3d1b7d2c4f2b957f1561ef0d4ae1', size: this.size, iconPack: this.iconPack, onClick: () => this.toggleDebugger(), icon: this.iconPack.debug, label: this.t('Troubleshooting'), variant: this.variant })));
    }
    static get style() { return DyteDebuggerToggleStyle0; }
}, [1, "dyte-debugger-toggle", {
        "variant": [513],
        "meeting": [16],
        "states": [16],
        "t": [16],
        "iconPack": [16],
        "size": [513]
    }]);
__decorate([
    SyncWithStore()
], DyteDebuggerToggle$1.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteDebuggerToggle$1.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], DyteDebuggerToggle$1.prototype, "t", void 0);
__decorate([
    SyncWithStore()
], DyteDebuggerToggle$1.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteDebuggerToggle$1.prototype, "size", void 0);
function defineCustomElement$1() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-debugger-toggle", "dyte-controlbar-button", "dyte-icon", "dyte-spinner"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-debugger-toggle":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteDebuggerToggle$1);
            }
            break;
        case "dyte-controlbar-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dyte-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dyte-spinner":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
    } });
}
defineCustomElement$1();

const DyteDebuggerToggle = DyteDebuggerToggle$1;
const defineCustomElement = defineCustomElement$1;

export { DyteDebuggerToggle, defineCustomElement };
