// @dytesdk/ui-kit/dist/components/p-2eccc6c6.js
import { p as proxyCustomElement, H, d as createEvent, h, e as Host } from './p-1d5392d7.js';
import { a as defaultIconPack, c as useLanguage } from './p-dce5d30e.js';
import { S as SyncWithStore } from './p-9ae68517.js';
import { d as defineCustomElement$1 } from './p-e7892c80.js';

const dyteMuteAllConfirmationCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:block;-webkit-user-select:none;-moz-user-select:none;user-select:none}.leave-modal{position:relative;display:flex;flex-direction:column;border-radius:var(--dyte-border-radius-md, 8px);--tw-bg-opacity:1;background-color:rgba(var(--dyte-colors-background-900, 26 26 26) / var(--tw-bg-opacity));padding:var(--dyte-space-4, 16px);color:rgb(var(--dyte-colors-text-1000, 255 255 255))}.leave-modal .header h2{margin:var(--dyte-space-0, 0px);margin-bottom:var(--dyte-space-3, 12px)}.leave-modal .content{font-size:14px}.leave-message p{margin-top:var(--dyte-space-3, 12px);margin-bottom:var(--dyte-space-3, 12px)}.leave-meeting{display:flex;flex-direction:row;justify-content:space-between;gap:var(--dyte-space-2, 8px)}.leave-meeting dyte-button{flex:1 1 0%}.end-meeting{margin-top:var(--dyte-space-2, 8px)}.message{margin-bottom:var(--dyte-space-4, 16px)}label{display:flex;align-items:center}input[type='checkbox']{margin-right:var(--dyte-space-2, 8px);accent-color:rgb(var(--dyte-colors-brand-500, 33 96 253))}";
const DyteMuteAllConfirmationStyle0 = dyteMuteAllConfirmationCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteMuteAllConfirmation = /*@__PURE__*/ proxyCustomElement(class DyteMuteAllConfirmation extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        this.stateUpdate = createEvent(this, "dyteStateUpdate", 7);
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        this.allowUnmute = true;
        this.onClose = () => {
            this.stateUpdate.emit({ activeMuteAllConfirmation: false });
        };
        this.onMuteAll = () => {
            var _a;
            (_a = this.meeting) === null || _a === void 0 ? void 0 : _a.participants.disableAllAudio(this.allowUnmute);
            this.onClose();
        };
    }
    render() {
        return (h(Host, { key: '8d447912a33466ac84c7104e3731e67e5d574201' }, h("div", { key: 'f36495740298c71957ecb4038ea11baa96150427', class: "leave-modal" }, h("div", { key: '5c7d23277e2fb5300b4d34800eb62e7a93e431dc', class: "header" }, h("h2", { key: 'bb08708a5d462763d51228e5368389c848f770b1', class: "title" }, this.t('mute_all.header'))), h("p", { key: '34273d1a512336de748ce1de4b498baebd2c7cfb', class: "message" }, this.t('mute_all.description')), h("div", { key: '2fae2f55acb20556e3478c5a57c9d3f5d0dae5ab', class: "content" }, h("div", { key: '67308bf61bfa8891e0193033779ab4952a95a8e1', class: "leave-meeting" }, h("dyte-button", { key: '03835bed6fe212b2d893ebd6c1d1061b6ef548e4', variant: "secondary", title: this.t('close'), onClick: this.onClose }, this.t('cancel')), h("dyte-button", { key: 'e9f5a1ef27c92446d6955e4898757070d39629ce', variant: "danger", title: this.t('mute_all'), onClick: this.onMuteAll }, this.t('mute_all')))))));
    }
    static get style() { return DyteMuteAllConfirmationStyle0; }
}, [1, "dyte-mute-all-confirmation", {
        "meeting": [16],
        "states": [16],
        "iconPack": [16],
        "t": [16],
        "allowUnmute": [32]
    }]);
__decorate([
    SyncWithStore()
], DyteMuteAllConfirmation.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteMuteAllConfirmation.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], DyteMuteAllConfirmation.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteMuteAllConfirmation.prototype, "t", void 0);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-mute-all-confirmation", "dyte-button"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-mute-all-confirmation":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteMuteAllConfirmation);
            }
            break;
        case "dyte-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}
defineCustomElement();

export { DyteMuteAllConfirmation as D, defineCustomElement as d };
