// @dytesdk/ui-kit/dist/components/p-a443344b.js
import { p as proxyCustomElement, H, h, e as Host } from './p-1d5392d7.js';
import { S as SyncWithStore } from './p-9ae68517.js';

const dyteIconCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:block;height:var(--dyte-space-10, 40px);outline:2px solid transparent;outline-offset:2px}:host .icon-wrapper{display:flex;height:100%;width:100%;flex-direction:column;align-items:center}:host svg{height:100%;width:100%;outline:2px solid transparent;outline-offset:2px}:host([size='xl']){box-sizing:border-box;display:block;height:var(--dyte-space-16, 64px) !important;width:var(--dyte-space-16, 64px) !important;--dyte-spinner-color:currentColor}:host([size='md']){box-sizing:border-box;display:block;height:var(--dyte-space-6, 24px) !important;width:var(--dyte-space-6, 24px) !important;--dyte-spinner-color:currentColor}:host([size='sm']){box-sizing:border-box;display:block;height:var(--dyte-space-4, 16px) !important;width:var(--dyte-space-4, 16px) !important;--dyte-spinner-color:currentColor}:host([variant='secondary']) .icon-wrapper{--tw-text-opacity:1;color:rgba(var(--dyte-colors-background-600, 60 60 60) / var(--tw-text-opacity))}:host([variant='danger']) .icon-wrapper{--tw-text-opacity:1;color:rgba(var(--dyte-colors-danger, 255 45 45) / var(--tw-text-opacity))}";
const DyteIconStyle0 = dyteIconCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const parseIcon = (icon) => {
    try {
        return JSON.parse(icon);
    }
    catch (e) {
        return icon;
    }
};
const DyteIcon = /*@__PURE__*/ proxyCustomElement(class DyteIcon extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Icon variant */
        this.variant = 'primary';
        /** Size */
        this.size = 'lg';
    }
    render() {
        return (h(Host, { key: '7c30027e6133e5ec64063771c08d37a75b5435bd' }, h("div", { key: '74cf660575c383f53e38ba4688612bc69dbe9b34', class: "icon-wrapper", innerHTML: parseIcon(this.icon), part: "wrapper" })));
    }
    static get style() { return DyteIconStyle0; }
}, [1, "dyte-icon", {
        "icon": [1],
        "variant": [513],
        "size": [513]
    }]);
__decorate([
    SyncWithStore()
], DyteIcon.prototype, "size", void 0);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteIcon);
            }
            break;
    } });
}
defineCustomElement();

export { DyteIcon as D, defineCustomElement as d };
