// @dytesdk/ui-kit/dist/components/p-b682daa9.js
import { p as proxyCustomElement, H, h } from './p-1d5392d7.js';
import { d as defaultConfig, a as defaultIconPack, c as useLanguage } from './p-dce5d30e.js';
import './p-e497ad6d.js';
import { S as SyncWithStore } from './p-9ae68517.js';
import { d as defineCustomElement$5 } from './p-b503759c.js';
import { d as defineCustomElement$4 } from './p-e7892c80.js';
import { d as defineCustomElement$3 } from './p-a443344b.js';
import { d as defineCustomElement$2 } from './p-cba97631.js';
import { d as defineCustomElement$1 } from './p-b4f8107f.js';

const dyteParticipantsStageQueueCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{margin-top:var(--dyte-space-2, 8px);display:flex;width:100%;flex-direction:column;font-size:14px}.stage-requested-participants{margin-bottom:var(--dyte-space-8, 32px)}.stage-requested-participants .bulk-actions{display:flex;gap:var(--dyte-space-2, 8px)}.stage-requested-participants .bulk-actions .accept-all-button{--tw-text-opacity:1;color:rgba(var(--dyte-colors-success, 98 165 4) / var(--tw-text-opacity))}.stage-requested-participants .bulk-actions dyte-button{flex:1 1 0%}h3,.heading-count{margin:var(--dyte-space-0, 0px);display:flex;align-items:center;justify-content:center;padding:var(--dyte-space-0, 0px);font-size:16px;font-weight:400;color:rgb(var(--dyte-colors-text-900, 255 255 255 / 0.88));text-align:center}.heading-count{font-size:14px}.participants{margin-top:var(--dyte-space-2, 8px);padding:var(--dyte-space-0, 0px)}.waiting-participant{margin-top:var(--dyte-space-2, 8px);margin-bottom:var(--dyte-space-2, 8px);display:flex;align-items:center}.waiting-participant .participant-details{margin-right:auto;display:flex;align-items:center}.waiting-participant .participant-details dyte-avatar{margin-right:var(--dyte-space-2, 8px);height:var(--dyte-space-8, 32px);width:var(--dyte-space-8, 32px);flex-shrink:0;font-size:14px}.waiting-participant .participant-details .name{overflow:hidden;text-overflow:ellipsis;white-space:nowrap;color:rgb(var(--dyte-colors-text-900, 255 255 255 / 0.88))}@media (min-width: 1080px){.waiting-participant .participant-details .name{max-width:var(--dyte-space-40, 160px)}}.waiting-participant .waitlist-controls{display:flex}.waiting-participant .waitlist-controls dyte-button{margin-left:var(--dyte-space-2, 8px);cursor:pointer;border-radius:var(--dyte-border-radius-sm, 4px)}.waiting-participant .waitlist-controls dyte-icon.accept{--tw-text-opacity:1;color:rgba(var(--dyte-colors-success, 98 165 4) / var(--tw-text-opacity))}.waiting-participant .waitlist-controls dyte-icon.deny{--tw-text-opacity:1;color:rgba(var(--dyte-colors-danger, 255 45 45) / var(--tw-text-opacity))}";
const DyteParticipantsStageQueueStyle0 = dyteParticipantsStageQueueCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteParticipantsStaged = /*@__PURE__*/ proxyCustomElement(class DyteParticipantsStaged extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Config */
        this.config = defaultConfig;
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** View mode for participants list */
        this.view = 'sidebar';
        /** Language */
        this.t = useLanguage();
        this.stageRequestedParticipants = [];
        this.updateStageRequestedParticipants = () => {
            this.stageRequestedParticipants = this.meeting.participants.joined
                .toArray()
                .filter((p) => p.stageStatus === 'REQUESTED_TO_JOIN_STAGE');
        };
        this.acceptStageRequest = async (p) => {
            const { userId } = p;
            await this.meeting.stage.grantAccess([userId]);
            this.updateStageRequestedParticipants();
        };
        this.rejectStageRequest = async (p) => {
            const { userId } = p;
            await this.meeting.stage.denyAccess([userId]);
            this.updateStageRequestedParticipants();
        };
        this.acceptAllStageRequest = async () => {
            await this.meeting.stage.grantAccess(this.stageRequestedParticipants.map((p) => p.userId));
            this.updateStageRequestedParticipants();
        };
        this.denyAllStageRequest = async () => {
            var _a;
            await ((_a = this.meeting.stage) === null || _a === void 0 ? void 0 : _a.denyAccess(this.stageRequestedParticipants.map((p) => p.userId)));
            this.updateStageRequestedParticipants();
        };
        this.shouldShowStageRequests = () => {
            return (this.meeting.self.permissions.stageEnabled &&
                this.meeting.self.permissions.acceptStageRequests &&
                this.stageRequestedParticipants.length > 0);
        };
        this.updateRequestList = async (stageRequests) => {
            var _a, _b, _c, _d, _e, _f, _g;
            if (!this.meeting.self.permissions.acceptStageRequests ||
                !this.meeting.self.permissions.stageEnabled) {
                this.stageRequestedParticipants = [];
                return;
            }
            if (this.meeting.meta.viewType === 'LIVESTREAM' ||
                ((_c = (_b = (_a = this.meeting) === null || _a === void 0 ? void 0 : _a.self) === null || _b === void 0 ? void 0 : _b.permissions) === null || _c === void 0 ? void 0 : _c.mediaRoomType) === 'HIVE') {
                if (!stageRequests) {
                    stageRequests = (_f = (_e = (await ((_d = this.meeting.stage) === null || _d === void 0 ? void 0 : _d.getAccessRequests()))) === null || _e === void 0 ? void 0 : _e.stageRequests) !== null && _f !== void 0 ? _f : [];
                }
                /**
                 * NOTE(ishita1805): Temporarily mapping `displayName` to `name` till socket service sends the correct key.
                 */
                this.stageRequestedParticipants = stageRequests.map((p) => {
                    return Object.assign(Object.assign({}, p), { name: p.displayName });
                });
            }
            else {
                this.stageRequestedParticipants = (_g = [
                    this.meeting.self,
                    ...this.meeting.participants.joined.toArray(),
                ]) === null || _g === void 0 ? void 0 : _g.filter((p) => p.stageStatus === 'REQUESTED_TO_JOIN_STAGE');
            }
        };
    }
    connectedCallback() {
        this.meetingChanged(this.meeting);
    }
    disconnectedCallback() {
        const { stage } = this.meeting;
        stage === null || stage === void 0 ? void 0 : stage.removeListener('stageAccessRequestUpdate', this.updateRequestList);
    }
    meetingChanged(meeting) {
        var _a;
        if (meeting == null)
            return;
        this.updateRequestList();
        (_a = meeting.stage) === null || _a === void 0 ? void 0 : _a.on('stageAccessRequestUpdate', this.updateRequestList);
    }
    render() {
        if (this.view !== 'sidebar' || !this.shouldShowStageRequests())
            return;
        return (h("div", { class: "stage-requested-participants" }, h("div", { class: "heading-count", part: "staged-heading-count" }, this.t('stage_request.header_title'), " (", this.stageRequestedParticipants.length, ")"), h("ul", { class: "participants", part: "staged-participants" }, this.stageRequestedParticipants.map((participant) => (h("li", { class: "waiting-participant", key: participant.id }, h("div", { class: "participant-details" }, h("dyte-avatar", { participant: participant, size: "sm", iconPack: this.iconPack, t: this.t }), h("p", { class: "name", title: participant.name }, participant.name)), h("div", { class: "waitlist-controls" }, h("dyte-tooltip", { label: this.t('stage_request.deny_request'), variant: "secondary" }, h("dyte-button", { variant: "secondary", kind: "icon", onClick: () => this.rejectStageRequest(participant) }, h("dyte-icon", { class: "deny", icon: this.iconPack.dismiss }))), h("dyte-tooltip", { label: this.t('stage_request.accept_request'), variant: "secondary" }, h("dyte-button", { variant: "secondary", kind: "icon", onClick: () => this.acceptStageRequest(participant) }, h("dyte-icon", { class: "accept", icon: this.iconPack.checkmark })))))))), h("div", { class: "bulk-actions" }, h("dyte-button", { class: "accept-all-button", variant: "secondary", onClick: this.acceptAllStageRequest }, this.t('stage_request.accept_all')), h("dyte-button", { class: "deny-all-button", variant: "danger", onClick: this.denyAllStageRequest }, this.t('stage_request.deny_all')))));
    }
    static get watchers() { return {
        "meeting": ["meetingChanged"]
    }; }
    static get style() { return DyteParticipantsStageQueueStyle0; }
}, [1, "dyte-participants-stage-queue", {
        "meeting": [16],
        "config": [16],
        "size": [513],
        "iconPack": [16],
        "view": [1],
        "t": [16],
        "stageRequestedParticipants": [32]
    }, undefined, {
        "meeting": ["meetingChanged"]
    }]);
__decorate([
    SyncWithStore()
], DyteParticipantsStaged.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteParticipantsStaged.prototype, "size", void 0);
__decorate([
    SyncWithStore()
], DyteParticipantsStaged.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteParticipantsStaged.prototype, "t", void 0);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-participants-stage-queue", "dyte-avatar", "dyte-button", "dyte-icon", "dyte-spinner", "dyte-tooltip"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-participants-stage-queue":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteParticipantsStaged);
            }
            break;
        case "dyte-avatar":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dyte-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dyte-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dyte-spinner":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "dyte-tooltip":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}
defineCustomElement();

export { DyteParticipantsStaged as D, defineCustomElement as d };
