// @dytesdk/ui-kit/dist/components/p-f3ee4c1f.js
import { p as proxyCustomElement, H, h, e as Host } from './p-1d5392d7.js';
import { a as defaultIconPack, c as useLanguage } from './p-dce5d30e.js';
import './p-e497ad6d.js';
import { S as SyncWithStore } from './p-9ae68517.js';
import { d as defineCustomElement$c } from './p-b503759c.js';
import { d as defineCustomElement$b } from './p-e7892c80.js';
import { d as defineCustomElement$a } from './p-d962d0d6.js';
import { d as defineCustomElement$9 } from './p-07fba69f.js';
import { d as defineCustomElement$8 } from './p-a443344b.js';
import { d as defineCustomElement$7 } from './p-b31e4510.js';
import { d as defineCustomElement$6 } from './p-80ecfb3b.js';
import { d as defineCustomElement$5 } from './p-9ae9dd32.js';
import { d as defineCustomElement$4 } from './p-3a600290.js';
import { d as defineCustomElement$3 } from './p-43dce699.js';
import { d as defineCustomElement$2 } from './p-cba97631.js';
import { d as defineCustomElement$1 } from './p-6deb1429.js';

const dyteChatSearchResultsCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:flex;height:100%;flex-direction:column;position:relative;--tw-bg-opacity:1;background-color:rgba(var(--dyte-colors-background-900, 26 26 26) / var(--tw-bg-opacity))}";
const DyteChatSearchResultsStyle0 = dyteChatSearchResultsCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteChatSearchResults = /*@__PURE__*/ proxyCustomElement(class DyteChatSearchResults extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        this.pageSize = 50;
        this.searchMessages = async (timestamp, size, reversed) => {
            return this.meeting.chat.searchMessages(this.query, {
                channelId: this.channelId,
                timestamp,
                size,
                reversed,
            });
        };
        this.nodeRenderer = (messages) => {
            return messages.map((message) => (h("dyte-chat-message", { key: message.id, message: message, disableControls: true })));
        };
    }
    render() {
        return (h(Host, { key: '984a388e6c194e27c5cb7918806bd9f156220ec7' }, h("dyte-paginated-list", { key: '444fa000de6428b4c0c6e0c829eb4ef944d2a211', pageSize: this.pageSize, pagesAllowed: 3, fetchData: this.searchMessages, createNodes: this.nodeRenderer, selectedItemId: this.query })));
    }
    static get style() { return DyteChatSearchResultsStyle0; }
}, [1, "dyte-chat-search-results", {
        "meeting": [16],
        "query": [1],
        "channelId": [1, "channel-id"],
        "iconPack": [16],
        "t": [16]
    }]);
__decorate([
    SyncWithStore()
], DyteChatSearchResults.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteChatSearchResults.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteChatSearchResults.prototype, "t", void 0);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["dyte-chat-search-results", "dyte-avatar", "dyte-button", "dyte-chat-message", "dyte-file-message", "dyte-icon", "dyte-image-message", "dyte-menu", "dyte-menu-item", "dyte-menu-list", "dyte-paginated-list", "dyte-spinner", "dyte-text-message"];
    components.forEach(tagName => { switch (tagName) {
        case "dyte-chat-search-results":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, DyteChatSearchResults);
            }
            break;
        case "dyte-avatar":
            if (!customElements.get(tagName)) {
                defineCustomElement$c();
            }
            break;
        case "dyte-button":
            if (!customElements.get(tagName)) {
                defineCustomElement$b();
            }
            break;
        case "dyte-chat-message":
            if (!customElements.get(tagName)) {
                defineCustomElement$a();
            }
            break;
        case "dyte-file-message":
            if (!customElements.get(tagName)) {
                defineCustomElement$9();
            }
            break;
        case "dyte-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$8();
            }
            break;
        case "dyte-image-message":
            if (!customElements.get(tagName)) {
                defineCustomElement$7();
            }
            break;
        case "dyte-menu":
            if (!customElements.get(tagName)) {
                defineCustomElement$6();
            }
            break;
        case "dyte-menu-item":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "dyte-menu-list":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "dyte-paginated-list":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "dyte-spinner":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "dyte-text-message":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}
defineCustomElement();

export { DyteChatSearchResults as D, defineCustomElement as d };
