// @dytesdk/ui-kit/dist/esm/dyte-leave-button.entry.js
import { r as registerInstance, c as createEvent, h, H as Host } from './index-08f4416b.js';
import { a as defaultIconPack, c as useLanguage } from './ui-store-9dd5dd86.js';
import { S as SyncWithStore } from './index-39fa23f1.js';

const dyteLeaveButtonCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:block}";
const DyteLeaveButtonStyle0 = dyteLeaveButtonCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteLeaveButton = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        this.stateUpdate = createEvent(this, "dyteStateUpdate", 7);
        /** Variant */
        this.variant = 'button';
        /** Icon pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
        this.leave = () => {
            this.stateUpdate.emit({ activeLeaveConfirmation: true });
        };
    }
    render() {
        const text = this.t('leave');
        return (h(Host, { key: '03c911ea8de75bef21b720f64b270c06193fb35c', label: text }, h("dyte-controlbar-button", { key: '9e0909fe191a3fd257d818054656fb65a318ad44', size: this.size, iconPack: this.iconPack, class: "leave red-icon", onClick: this.leave, icon: this.iconPack.call_end, label: text, variant: this.variant, part: "controlbar-button" })));
    }
};
__decorate([
    SyncWithStore()
], DyteLeaveButton.prototype, "size", void 0);
__decorate([
    SyncWithStore()
], DyteLeaveButton.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteLeaveButton.prototype, "t", void 0);
DyteLeaveButton.style = DyteLeaveButtonStyle0;

export { DyteLeaveButton as dyte_leave_button };
