// @dytesdk/ui-kit/dist/esm/dyte-header.entry.js
import { r as registerInstance, h, H as Host } from './index-08f4416b.js';
import { d as defaultConfig, a as defaultIconPack, c as useLanguage } from './ui-store-9dd5dd86.js';
import { R as Render } from './index-e2bee3ae.js';
import { S as SyncWithStore } from './index-39fa23f1.js';

const dyteHeaderCss = ":host{line-height:initial;font-family:var(--dyte-font-family, sans-serif);font-feature-settings:normal;font-variation-settings:normal}p{margin:var(--dyte-space-0, 0px);padding:var(--dyte-space-0, 0px)}:host{display:flex;height:var(--dyte-space-12, 48px);align-items:center;--tw-bg-opacity:1;background-color:rgba(var(--dyte-colors-background-1000, 8 8 8) / var(--tw-bg-opacity));padding-left:var(--dyte-space-4, 16px);padding-right:var(--dyte-space-4, 16px)}@media only screen and (max-device-height: 480px) and (orientation: landscape){:host{display:none !important}}@media only screen and (max-height: 480px) and (orientation: landscape){:host{display:none !important}}";
const DyteHeaderStyle0 = dyteHeaderCss;

var __decorate = (undefined && undefined.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function")
        r = Reflect.decorate(decorators, target, key, desc);
    else
        for (var i = decorators.length - 1; i >= 0; i--)
            if (d = decorators[i])
                r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
const DyteHeader = class {
    constructor(hostRef) {
        registerInstance(this, hostRef);
        /** Variant */
        this.variant = 'solid';
        /** Whether to render the default UI */
        this.disableRender = false;
        /** Config */
        this.config = defaultConfig;
        /** Icon Pack */
        this.iconPack = defaultIconPack;
        /** Language */
        this.t = useLanguage();
    }
    render() {
        const defaults = {
            meeting: this.meeting,
            config: this.config,
            states: this.states,
            t: this.t,
            iconPack: this.iconPack,
            size: this.size,
        };
        return (h(Host, { key: '8c6d62fc6a63020885d9e367b281d584e4f86580' }, !this.disableRender && h(Render, { key: '9e0ce3f2e2eb8bebe322591cedbb243b588636ce', element: "dyte-header", defaults: defaults, onlyChildren: true }), h("slot", { key: '7eef46d4d0c1bae48b3fe852cd8347aa33aa60dc' })));
    }
};
__decorate([
    SyncWithStore()
], DyteHeader.prototype, "meeting", void 0);
__decorate([
    SyncWithStore()
], DyteHeader.prototype, "states", void 0);
__decorate([
    SyncWithStore()
], DyteHeader.prototype, "iconPack", void 0);
__decorate([
    SyncWithStore()
], DyteHeader.prototype, "t", void 0);
__decorate([
    SyncWithStore()
], DyteHeader.prototype, "size", void 0);
DyteHeader.style = DyteHeaderStyle0;

export { DyteHeader as dyte_header };
